packages/clerk-js/src/utils/authPropHelpers.ts

```diff
@@ -29,15 +30,27 @@ type PickRedirectionOptions = {
  */
 export const pickRedirectionProp = (
   key: PickRedirectionUrlKey,
-  { ctx, queryParams, displayConfig, options }: PickRedirectionOptions,
+  { ctx, queryParams, displayConfig, options, allowedRedirectOrigins }: PickRedirectionOptions,
   accessRedirectUrl = true,
 ): string => {
   const snakeCaseField = camelToSnake(key);
   const queryParamValue = queryParams?.[snakeCaseField];
 
+  const primaryQueryParamRedirectUrl = typeof queryParamValue === 'string' ? queryParamValue : null;
+  const secondaryQueryParamRedirectUrl =
+    accessRedirectUrl && typeof queryParams?.redirect_url === 'string' ? queryParams.redirect_url : null;
+
+  let queryParamUrl: string | null | undefined = primaryQueryParamRedirectUrl || secondaryQueryParamRedirectUrl;
+
+  if (allowedRedirectOrigins) {
+    queryParamUrl = getFirstAllowedRedirectAndWarn(
+      [primaryQueryParamRedirectUrl, secondaryQueryParamRedirectUrl].filter(i => i !== null) as string[],
+      allowedRedirectOrigins,
+    );
+  }
```

Comment:
❓ why not move the `allowedRedirectOrigins` check after the calculation and validation of the `url` ?